dash.el

```diff
@@ -1054,11 +1053,9 @@ element of LIST paired with the unmodified element of LIST."
 (defun dash--table-carry (lists restore-lists &optional re)
   "Helper for `-table' and `-table-flat'.
 
-If a list overflows, carry to the right and reset the list.
-
-Return how many lists were re-seted."
-  (while (and (not (car lists))
-              (not (equal lists '(nil))))
+If a list overflows, carry to the right and reset the list."
```

Comment:
This seems useless.  I much prefer to read conditions as "this and that" then "not this or that", especially if I'm expecting a positive answer.